pkg/operator/controllers/base/aro_controller.go

```diff
@@ -33,6 +38,27 @@ func (c *AROController) GetName() string {
 	return c.Name
 }
 
+func (c *AROController) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	cluster, err := c.GetCluster(ctx)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Controller can be disabled if it defines an "enabled" operator flag.
+	if c.EnabledFlag != "" && !cluster.Spec.OperatorFlags.GetSimpleBoolean(c.EnabledFlag) {
+		c.Log.Debug("controller is disabled")
+		return c.Reconciler.ReconcileDisabled(ctx, request, cluster)
+	}
+
+	c.Log.Debug("running")
+	return c.Reconciler.ReconcileEnabled(ctx, request, cluster)
+}
+
+func (c *AROController) ReconcileDisabled(ctx context.Context, request ctrl.Request, cluster *arov1alpha1.Cluster) (ctrl.Result, error) {
+	c.ClearConditions(ctx)
```

Comment:
Do we want to clear out the current conditions when disabling? I'm wondering if that could cause issues with debugging, in that we might want to keep the conditions around after it's disabled. I don't know of any reason why we couldn't leave the conditions as they are if there's an enabled/disabled condition?